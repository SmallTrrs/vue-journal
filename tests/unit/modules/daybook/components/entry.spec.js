import { shallowMount } from '@vue/test-utils'
import EntryComp from '@/modules/daybook/components/EntryComp'
import { journalState} from '../../../mock-data/test-journal-state'

describe('Pruebas Entry Component', () =>{

   const mockRouter = {
      push: jest.fn()
   }

   const wrapper = shallowMount(EntryComp, {
      props: {
         entry: journalState.entries[0]
      },
      global:{
         mocks:{
            $router: mockRouter
         }
      }
   })

   test('Debe hacer match con el snapshot', ()=>{
      
      expect( wrapper.html() ).toMatchSnapshot()

   })

   test('Debe de redireccionar al hacer click en el entry-container', ()=>{
      
      const entryContainer = wrapper.find('.entry-container')

      entryContainer.trigger('click')

      expect( mockRouter.push ).toHaveBeenCalledWith({
         name: "entry", params: { id: "-MzcI73tkrugpaWu6vh6"}
      })

   })

   test('pruebas en las propiedades computadas', ()=>{
 
   expect(wrapper.vm.day ).toBe(1)
   expect(wrapper.vm.month ).toBe('Abril')
   expect(wrapper.vm.yearDay ).toBe('2022, Viernes')

   })
})