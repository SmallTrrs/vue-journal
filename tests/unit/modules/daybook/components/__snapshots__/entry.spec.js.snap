// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas Entry Component Debe hacer match con el snapshot 1`] = `
<div class="mb-3 pointer p-2 entry-container">
  <!-- titulo  -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">1</span><span class="mx-1 fs-5">Abril</span><span class="mx-2 fw-light">2022, Viernes</span></div>
  <div class="entry-description">Primero de Abril</div>
</div>
`;
